test/functional/apps/discover/group1/_shared_links.ts

```diff
@@ -64,7 +65,16 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
       describe('permalink', function () {
         it('should allow for copying the snapshot URL', async function () {
+          if (await PageObjects.share.isShareModalOpen()) {
+            await PageObjects.share.closeShareModal();
+          }
+          if (!(await PageObjects.share.isShareMenuOpen())) {
+            await PageObjects.share.clickShareTopNavButton();
+          }
           const actualUrl = await PageObjects.share.getSharedUrl();
+          if (await PageObjects.share.isShareModalOpen()) {
+            await PageObjects.share.closeShareModal();
+          }
```

Comment:
Should these checks (before/after `getSharedUrl`) be moved into `PageObjects.share.getSharedUrl()`?

Reply:
I'm planning on doing test clean up after the next redesign since the UI interactions will be changing again. I like the idea of moving most of the test helpers into the share PageObjects for cleanliness.